File: src/main/java/upbrella/be/rent/dto/response/ReturnFormResponse.java
package upbrella.be.rent.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.rent.entity.History;
import upbrella.be.store.entity.StoreMeta;

@Getter
@Builder
public class ReturnFormResponse {

    private String classificationName;
    private String rentStoreName;
    private long storeId;

    public static ReturnFormResponse of(StoreMeta storeMeta, History history) {

        return ReturnFormResponse.builder()
                .classificationName(storeMeta.getClassification().getName())
                .rentStoreName(history.getRentStoreMeta().getName())
                .storeId(storeMeta.getId())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/rent/dto/response/ReturnFormResponse.kt
package upbrella.be.rent.dto.response

import upbrella.be.rent.entity.History
import upbrella.be.store.entity.StoreMeta

data class ReturnFormResponse(
    val classificationName: String,
    val rentStoreName: String,
    val storeId: Long
) {
    companion object {
        fun of(storeMeta: StoreMeta, history: History): ReturnFormResponse {
            return ReturnFormResponse(
                classificationName = storeMeta.classification!!.name!!,
                rentStoreName = history.rentStoreMeta.name,
                storeId = storeMeta.id!!
            )
        }
    }
}
